File: functions/api/src/routes/documentation.rs
use crate::{context::Context, routes::InputMetadata};
use lambda_runtime::Error;
use serde::Deserialize;
use serde_json::{Value, json};

#[derive(Deserialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct DocumentationRequest {
    #[serde(flatten)]
    metadata: InputMetadata,
    name: String,
}

pub async fn handle(req: DocumentationRequest) -> Result<Value, Error> {
    let Some(library_name) = req.metadata.library.as_deref() else {
        return Err(anyhow::format_err!("missing library").into());
    };

    let documentation = Context::shared()
        .compile_library(library_name)
        .await
        .ok()
        .and_then(|compiler| compiler.documentation(&req.name));

    Ok(json!({
        "documentation": documentation,
    }))
}

File: server/src/routes/documentation.ts
import { handler } from "../handler";
import { InputMetadata, libraries } from "../library";

const DocumentationRequest = InputMetadata;

export default handler(DocumentationRequest, async (body) => {
    if (body.library == null) {
        return { statusCode: 400, body: { error: "missing library" } };
    }

    const library = libraries[body.library]?.[0];
    if (library == null) {
        return { statusCode: 400, body: { error: "unknown library" } };
    }

    return { statusCode: 200, body: { items: library.docs } };
});
